articles/100_ament.md

```diff
@@ -20,34 +20,93 @@ author: '[Dirk Thomas](https://github.com/dirk-thomas)'
-  Currently it only contains a Python implementation but other languages like C++ should follow.
+Each package is build separately with its own build system.
+In order to make the output of one package available to other packages each package can extend the environment in a way that downstream packages can find and use its artifacts and resources.
+If the resulting artifacts are installed into e.g. `/usr` it might not be necessary to alter the environment at all since these folders are commonly being searched by various tools.
+
+
+### The ament command line tool
+
+[ament_tools](https://github.com/ament/ament_tools) is a Python package which provides the command line tool `ament` to build, test, install, and uninstall packages.
+It is similar to [catkin_tools](https://github.com/catkin/catkin_tools) and builds each package in a workspace in topological order.
+Support for different build systems is integrated through extension points.
+
+While it currently does not build packages in parallel that feature will be added in the future to speed up the build process.
+The goal is to reuse common functionality from catkin_tools by making it available through a third package which can be used by both tools.
 
 
 ### Integrate arbitrary build systems
 
-Each package can utilize a different build system to perform the steps of configuring, building and installing.
+Each package can utilize a different build system to perform the steps of configuring, building, and installing.
+The build type is defined in each package manifest using the [build_type](http://www.ros.org/reps/rep-0140.html#build-type) tag in the export section.
+
 Currently supported are CMake and Python but support for others (e.g. autotools, plain Makefiles) will likely be added in the future.
-For each of them the native steps are being applied separately by the ament tools.
+For each of them the native steps are being applied separately by the ament tool.
+
+
+#### Build type: ament_cmake
+
+The CMake project `ament_cmake` provides several convenience functions to make it easier to write CMake-based packages:
+
+* It generate a CMake config file for the package.
+  That allows passing information (e.g. about include directories and libraries) to downstream packages.
+  Additionally it makes it easy to pass along information from recursive dependencies (and e.g. takes care about the ordering of include directories).
+* It provides an easy interface to register tests and ensure that JUnit-compatible result files are generated for those.
+  Currently it supports a few different testing frameworks like `nosetests`, `gtest`, and `gmock`.
+* It allows to generate environment hook to extend the environment e.g. by extending the `PATH`.
+* It provides a CMake API to read and write the [ament resource index](https://github.com/ament/ament_cmake/blob/master/ament_cmake_core/doc/resource_index.md).
```

Comment:
`to read and write to the ...` or `to read and write [ament resource index](...) entries.`